sequencer/src/block/full_payload/ns_proof.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{
+    block::{
+        full_payload::{NsIndex, NsTable, Payload, PayloadByteLen},
+        namespace_payload::NsPayloadOwned,
+    },
+    NamespaceId, Transaction,
+};
+use hotshot_types::{
+    traits::EncodeBytes,
+    vid::{vid_scheme, LargeRangeProofType, VidCommitment, VidCommon, VidSchemeType},
+};
+use jf_vid::{
+    payload_prover::{PayloadProver, Statement},
+    VidScheme,
+};
+use serde::{Deserialize, Serialize};
+
+/// Proof of correctness for namespace payload bytes in a block.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct NsProof {
+    ns_index: NsIndex,
+    ns_payload: NsPayloadOwned,
+    ns_proof: Option<LargeRangeProofType>, // `None` if ns_payload is empty
+}
+
+impl NsProof {
+    /// Returns the payload bytes for namespace `ns_id`, along with a proof of
+    /// correctness for those bytes. Returns `None` if `ns_id` is not in the
+    /// namespace table, or on error.
```

Comment:
This doc was out of date. Fixed in https://github.com/EspressoSystems/espresso-sequencer/pull/1499/commits/65b6f62cfedb4a47544a652b788ad8fd04e20966

This method actually takes a `NsIndex` arg instead of a `NamespaceId` arg. An out-of-bounds index is viewed as an error, so this method actually returns `None` only on error. (We could change error handling to use `Result` here but I'm not aware that we've settled on a good error handling pattern so for now I'm continuing to live in denial and return an `Option` instead.)